.dockerignore

```diff
@@ -0,0 +1,4 @@
+/target/
```

Comment:
This duplicates a lot of what is in the `.gitignore`. Can we avoid having an extra file, and somehow reuse the `.gitignore`?

Reply:
i think docker looks at dockerignore and git looks at gitignore. Never seen one use the other

Reply:
What about making `.dockerignore` a symlink to `.gitignore`? That would keep them in sync.

Reply:
We don't want them to be the exact same file. While they're similar we want to dockerignore `.git` and other extra files.

Reply:
Gotcha, that makes sense, sounds good.